drivers/mtd/mtd_nandram.c

```diff
@@ -0,0 +1,473 @@
+  uint8_t ecc_4;
+  uint8_t ecc_5;
+
+  /* Using reserved (8 bytes) */
+
+  uint16_t n_read;
+  uint16_t n_write; /* 10 */
+  uint16_t n_erase;
+  uint8_t  free;    /* Erased page: NAND_RAM_PAGE_* */
+  uint8_t  __res2;
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static uint64_t                nand_ram_ins_i = 0; /* Instruction counter */
+static mutex_t                 nand_ram_dev_mut;
+static struct nand_ram_data_s  nand_ram_flash_data[NAND_RAM_N_PAGES];
+static struct nand_ram_spare_s nand_ram_flash_spare[NAND_RAM_N_PAGES];
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * External Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nand_ram_storage_status
+ *
+ * Description:
+ *   Writes per-page status of virtual NAND Flash.
+ *
+ ****************************************************************************/
+
+static void nand_ram_storage_status(void)
+{
+  uint32_t  i;
+  uint16_t  reads;
+  uint16_t  writes;
+  uint16_t  erases;
+  uint8_t   bad;
+
+  /* Wear */
+
+  for (i = 0; i < NAND_RAM_N_PAGES; i++)
+    {
+      reads = nand_ram_flash_spare[i].n_read;
+      writes = nand_ram_flash_spare[i].n_write;
+      erases = nand_ram_flash_spare[i].n_erase;
+      bad = (nand_ram_flash_spare[i].bad != NAND_RAM_BLOCK_GOOD);
+
+      NAND_RAM_STATUS_LOG(
+        "Block %3d, Page %6d, Bad: %1d |"
+        " Reads: %6d, Writes: %6d, Erases: %6d\n",
+        i >> NAND_RAM_LOG_PAGES_PER_BLOCK, i, bad,
+        reads, writes, erases);
+    }
+
+  return;
+}
+
+static inline void nand_ram_status(void)
+{
+  if (nand_ram_ins_i % NAND_RAM_STATUS_LEVEL == 0)
+    {
+      nand_ram_storage_status();
+    }
+}
+
+/****************************************************************************
+ * Name: nand_ram_storage_init
+ *
+ * Description:
+ *   Initializes the actual NAND Device that is emulated from RAM.
+ *
+ ****************************************************************************/
+
+static void nand_ram_storage_init(void)
+{
+  int i;
+
+  /* Hard coded array for bad block indexes */
+
+  int nand_ram_rand_bad_blk_indx[] = {4, 14, 19, 21, 28, 30, 107, 108, 164,
```

Comment:
try this:
```suggestion
  int nand_ram_rand_bad_blk_indx[] =
  {
    4, 14, 19, 21, 28, 30, 107, 108, 164,
    ...
  };
```
BTW, it's better to move the const array to global section

Reply:
I'll give this a try thanks!